File: src/main/webapp/app/entities/reports/services/cash-flow-bfr.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpResponse } from '@angular/common/http';
import { Observable } from 'rxjs';
import { SERVER_API_URL } from 'app/app.constants';
import { IBfrEvolution, IBfrSnapshot } from 'app/shared/model/report';

@Injectable({ providedIn: 'root' })
export class CashFlowBfrService {
  private readonly resourceUrl = SERVER_API_URL + 'api/cash-flow-bfr';
  private readonly http = inject(HttpClient);

  getSnapshot(): Observable<HttpResponse<IBfrSnapshot>> {
    return this.http.get<IBfrSnapshot>(`${this.resourceUrl}/snapshot`, { observe: 'response' });
  }

  getEvolution(): Observable<HttpResponse<IBfrEvolution>> {
    return this.http.get<IBfrEvolution>(`${this.resourceUrl}/evolution`, { observe: 'response' });
  }
}

File: pharmaSmart-app/src/main/java/com/kobe/warehouse/service/report/CashFlowBfrService.java
package com.kobe.warehouse.service.report;

import com.kobe.warehouse.service.dto.report.BfrEvolutionDTO;
import com.kobe.warehouse.service.dto.report.BfrSnapshotDTO;

public interface CashFlowBfrService {

    BfrSnapshotDTO getSnapshot();

    BfrEvolutionDTO getEvolution();
}
